pkg/cluster/driver/capi/stage.go

```diff
@@ -58,10 +61,79 @@ func imageFromMachineTemplate(mt *unstructured.Unstructured) (*core.Image, error
 	return img, nil
 }
 
+// patchControlPlane adds any changes required to a KubeadmControlPlane
+func patchControlPlane(restConfig *rest.Config, kcp *unstructured.Unstructured) error {
+	// Ensure this is a KubeadmControlPlane
+	if kcp.GetAPIVersion() != capi.ControlPlaneAPI || kcp.GetKind() != capi.KubeadmControlPlane {
+		return fmt.Errorf("Control plane object %s in namespace %s is not a %s/%s", kcp.GetName(), kcp.GetNamespace(), capi.ControlPlaneAPI, capi.KubeadmControlPlane)
+	}
+
+	didUpdate := false
+	annots := kcp.GetAnnotations()
+	if annots == nil {
+		annots = map[string]string{}
+	}
+	_, ok := annots[capi.SkipKubeProxyAnnotation]
+	if !ok {
+		annots[capi.SkipKubeProxyAnnotation] = "true"
+		didUpdate = true
+	}
+
+	_, ok = annots[capi.SkipCoreDNSAnnotation]
+	if !ok {
+		annots[capi.SkipCoreDNSAnnotation] = "true"
+		didUpdate = true
+	}
+
+	if !didUpdate {
+		return nil
+	}
+
+	kcp.SetAnnotations(annots)
+	err := k8s.UpdateResource(restConfig, kcp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getControlPlanePatches calculates the set of patches that need to be
+// applied to the KubeadmControlPlane after staging is complete to
+// apply the new configuration
+func getControlPlanePatches(kcp *unstructured.Unstructured, version string, mtName string) *util.JsonPatches {
+	ret := &util.JsonPatches{}
+
+	// These are mandatory changes to update control plane nodes
+	ret.Replace(capi.ControlPlaneVersion, version)
+	ret.Replace(append(capi.ControlPlaneMachineTemplateInfrastructureRef, "name"), mtName)
+
+	//  The joinConfiguration needs to apply the OCK patches
+	patches, found, err := unstructured.NestedStringMap(kcp.Object, capi.ControlPlaneJoinPatches...)
+	if err != nil {
+		log.Warnf("Error getting KubeadmControlPlane.%s: %v", strings.Join(capi.ControlPlaneJoinPatches, "."), err)
+		return ret
```

Comment:
so this is not a fatal error?

Reply:
It's more that it can't happen.  In any case, I updated the function to return an error and check for that error when called.  Leaving the warning is gross.